File: apps/mobile/lib/services/shifts_service.dart
import 'api_service.dart';

class ShiftsService {
  static Future<Map<String, dynamic>> startShift({
    required String staffId,
    required String branchId,
    required int startingCash,
  }) async {
    return ApiService.post('/shifts/start', {
      'staffId': staffId,
      'branchId': branchId,
      'startingCash': startingCash,
    });
  }

  static Future<Map<String, dynamic>> closeShift({
    required String id,
    required int endingCash,
  }) async {
    return ApiService.post('/shifts/$id/close', {
      'endingCash': endingCash,
    });
  }

  static Future<Map<String, dynamic>?> getCurrentShift(String staffId) async {
    try {
      final result = await ApiService.get('/shifts/current?staffId=$staffId');
      return result;
    } catch (_) {
      return null;
    }
  }

  static Future<Map<String, dynamic>> getShiftSummary(String shiftId) async {
    return ApiService.get('/shifts/$shiftId/summary');
  }

  static Future<List<dynamic>> getAll({String? branchId, String? staffId}) async {
    final params = <String>[];
    if (branchId != null) params.add('branchId=$branchId');
    if (staffId != null) params.add('staffId=$staffId');
    final query = params.isNotEmpty ? '?${params.join('&')}' : '';
    return ApiService.getList('/shifts$query');
  }
}

File: apps/admin/src/services/shifts.service.js
import { apiClient } from '../lib/api-client';

export const shiftsService = {
    getAll: (filters = {}) => {
        const params = new URLSearchParams();
        if (filters.branchId) params.set('branchId', filters.branchId);
        if (filters.staffId) params.set('staffId', filters.staffId);
        const qs = params.toString();
        return apiClient.get(`/shifts${qs ? `?${qs}` : ''}`);
    },
    getCurrent: (staffId) => apiClient.get(`/shifts/current?staffId=${staffId}`),
    start: (data) => apiClient.post('/shifts/start', data),
    close: (id, endingCash) => apiClient.post(`/shifts/${id}/close`, { endingCash }),
    getSummary: (id) => apiClient.get(`/shifts/${id}/summary`),
};
